protocols/imap/src/main/java/org/apache/james/imap/api/ImapConfiguration.java

```diff
@@ -0,0 +1,85 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.imap.api;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.james.imap.processor.IdleProcessor;
+
+import com.google.common.collect.ImmutableSet;
+
+public class ImapConfiguration {
+    public static Builder builder() {
+        return new Builder();
+    }
+    public static class Builder {
+        private long idleTimeInterval;
+        private TimeUnit idleTimeIntervalUnit;
+        private Set<String> disabledCaps;
+
+        private Builder() {
+            this.idleTimeInterval = IdleProcessor.DEFAULT_HEARTBEAT_INTERVAL_IN_SECONDS;
```

Comment:
yes, because otherwise, impl reads conf and conf reads default value in impl.
If static fields didn't exist, it would not be possible at all.